repo/blob/storage.go

```diff
@@ -412,3 +413,15 @@ func ReadBlobMap(ctx context.Context, br Reader) (map[ID]Metadata, error) {
 
 	return blobMap, nil
 }
+
+// WithRequestTimeout adds a deadline to the context if RequestTimeout is greater than 0.
+// Returns the updated context and a cancel function that should be called to free resources.
+func WithRequestTimeout(ctx context.Context, seconds int64) (context.Context, context.CancelFunc) {
+	if seconds > 0 {
+		deadline := clock.Now().Add(time.Duration(seconds) * time.Second)
+		return context.WithDeadline(ctx, deadline)
```

Comment:
Please use `context.WithTimeout()`
```suggestion
		return context.WithTimeout(ctx, seconds*time.Second)
```

Reply:
Done